File: scala/src/test/scala/kata/PhotoSorterSpec.scala
package kata

import org.scalatest._

class PhotoSorterSpec extends FlatSpec with Matchers {
  val inputDir = "../validate/testData/input/"
  val outputDir = "../validate/target/"
  val currentDir = "../validate/testData/currentPhotos/"

  "The year-month string " should "be parsed correctly from file names" in {
    PhotoSorter.getYearMonthFromFileName("20170101_123456.jpg") shouldEqual Some("201701")
    PhotoSorter.getYearMonthFromFileName("IMG-20170101-123456.jpg") shouldEqual Some("201701")
    PhotoSorter.getYearMonthFromFileName("x/y/IMG-20170101-123456.jpg") shouldEqual Some("201701")
    PhotoSorter.getYearMonthFromFileName("IM-20170101-123456.jpg") shouldEqual None
    PhotoSorter.getYearMonthFromFileName("aap") shouldEqual None
  }

  "The number of files found in the test input folder" should "equal 5" in {
    PhotoSorter.findAllJpgFiles(inputDir).size shouldEqual 5
  }

  "3 files " should "be copied to 201701 if the current folder is not specified" in {
    PhotoSorter.copy(inputDir, outputDir)
    PhotoSorter.findAllJpgFiles(outputDir + "201701").size shouldEqual 3
  }

  "2 files " should "be copied to 201701 if the current folder is specified" in {
    PhotoSorter.copy(inputDir, outputDir, currentDir)
    PhotoSorter.findAllJpgFiles(outputDir + "201701").size shouldEqual 2
  }

  "2 files " should "be copied to 201702" in {
    PhotoSorter.copy(inputDir, outputDir)
    PhotoSorter.findAllJpgFiles(outputDir + "201702").size shouldEqual 2
  }
}

File: javascript/spec/kata/PhotoSorterSpec.js
describe("PhotoSorter", function () {
    var PhotoSorter = require('../../src/PhotoSorter.js');

    describe("PhotoSorter", function () {
        it("should parse date from fileName", function () {
            photoSorter = new PhotoSorter()
            expect(photoSorter.parseDateFromFileName('20170101_123456')).toEqual('201701');
            expect(photoSorter.parseDateFromFileName('IMG-20170101_123456')).toEqual('201701');
            expect(photoSorter.parseDateFromFileName('input/data/20170101_123456')).toEqual('201701');
            expect(photoSorter.parseDateFromFileName('IM-20170101_123456')).toEqual('');
            expect(photoSorter.parseDateFromFileName('aap')).toEqual('');
            expect(photoSorter.parseDateFromFileName('')).toEqual('');
        }),
            it("should find 5 jpg files in testData/input", function () {
                expect(photoSorter.getListOfJpgFiles('../validate/testData/input').length).toEqual(5);
            }),
            it("should copy 3 jpg files in /tmp/output/201701 and 2 in /tmp/output/201702 if currentDir is left empty", function () {
                photoSorter.copyFiles('../validate/testData/input', '/tmp/output/');
                expect(photoSorter.getListOfJpgFiles('/tmp/output/201701').length).toEqual(3);
                expect(photoSorter.getListOfJpgFiles('/tmp/output/201702').length).toEqual(2);
            }),
            it("should copy 2 jpg files in /tmp/output/201701 and 2 in /tmp/output/201702 if currentDir is set", function () {
                photoSorter.copyFiles('../validate/testData/input', '/tmp/output/', '../validate/testData/currentPhotos');
                expect(photoSorter.getListOfJpgFiles('/tmp/output/201701').length).toEqual(2);
                expect(photoSorter.getListOfJpgFiles('/tmp/output/201702').length).toEqual(2);
            });
    });
});
